tests/Kernel/LanguageSwitcherTest.php

```diff
@@ -15,6 +16,65 @@ class LanguageSwitcherTest extends MultilingualAbstractKernelTestBase {
    * Test language switcher link list rendering.
    */
   public function testLanguageSwitcherLinkListRendering(): void {
+    $this->assertDefaultLanguageBlock();
+
+    $lang_config = $this->container->get('config.factory')->get('language.negotiation');
+
+    // Create the Icelandic language.
+    $language = ConfigurableLanguage::createFromLangcode('is');
+    $language->setThirdPartySetting('oe_multilingual', 'category', 'non_eu');
+    $language->save();
+    // Load the language object.
+    $icelandic = $this->container->get('language_manager')->getLanguage('is');
+    $lang_id = $icelandic->getId();
+    $lang_prefix = $lang_config->get('url.prefixes.' . $lang_id);
+
+    // Build the language block.
+    $crawler = $this->renderLanguageBlock();
+
+    // Make sure that language switcher overlay is present.
+    $actual = $crawler->filter('.ecl-language-list--overlay');
+    $this->assertCount(1, $actual);
+
+    // Assert there is no Icelandic language in the EU category.
+    $actual = $crawler->filter(".ecl-language-list--overlay .ecl-language-list__eu a.ecl-language-list__link[lang={$lang_prefix}]");
+    $this->assertCount(0, $actual);
+
+    // Check for EU languages category.
+    $actual = $crawler->filter('.ecl-language-list__eu .ecl-language-list__category')->text();
+    $this->assertEquals('EU official languages', trim($actual));
+
+    // Check for non-EU languages category.
+    $actual = $crawler->filter('.ecl-language-list__non-eu .ecl-language-list__category')->text();
+    $this->assertEquals('Non-EU languages', trim($actual));
+
+    // Assert there is only one link in the non-EU category.
+    $actual = $crawler->filter('.ecl-language-list--overlay .ecl-language-list__non-eu a');
+    $this->assertCount(1, $actual);
+
+    // Assert the Icelandic language link.
+    $actual = $crawler->filter(".ecl-language-list--overlay .ecl-language-list__non-eu a.ecl-language-list__link[lang={$lang_prefix}]")->text();
+    $this->assertEquals(
+      $icelandic->getName(),
+      preg_replace('/[\x00-\x1F\x7F\xA0]/u', '', $actual)
```

Comment:
Why is this necessary?

Reply:
I did not create this, this code is from before. But it was done by Antonio to remove any non-printable characters from $actual.